src/pages/en/guides/markdown-content.md

```diff
@@ -103,14 +106,67 @@ An example blog post `content` object might look like:
 ```
 
 :::note
-`astro`, `file`, and `url` are the only guaranteed properties provided by Astro in the `content` prop. The rest of the object is defined by your frontmatter variables.
+`astro`, `file`, and `url` are the only guaranteed properties provided by Astro in the Markdown `content` prop. The rest of the object is defined by your frontmatter variables.
+
+📚 Read more about the [MDX content prop](/en/guides/integrations-guide/mdx/).
+
 :::
 
+#### Layouts for multiple page types
+
+An Astro layout can receive both the content object from `.md` and `.mdx` files, as well as any named props passed from `.astro` files.
+
+In the example below, the layout will display the page title either from an Astro component passing a `title` attribute, or from a frontmatter YAML `title` property:
+
+```astro
+---
+//src/components/MyLayout.astro
+const { content, title } = Astro.props
+
+const pageTitle = title || content.title
+---
+<html>
+  <head></head>
+  <body>
+    <h1>{pageTitle}</h1>
+    <slot />
+  </body>
+</html>
+```
+
+ALTERNATE EXAMPLE - CHOOSE WHICH ONE WE WANT, IF EITHER?
+
+```astro
+---
+// src/components/MyLayout.astro
+const { title } = Astro.props || Astro.props.content
+---
+<html>
+  <head></head>
+  <body>
+    <h1>{title}</h1>
+    <slot />
+  </body>
+</html>
+```
+
 ### Frontmatter as Props
```

Comment:
Now that I’m seeing this section, it’s not super clear _how_ these props get where they are in the example. Basically the pattern is grab `content` from a Markdown file (e.g. in a layout) then pass the object to another component.

I’m actually not 100% sure this section adds that much and is kind of confusing as-is. Do we want to consider removing it? Otherwise, I think we need an example where you see a layout render `<MyAstroComponent content={content} />`. Also given the md/mdx confusion do we need `content={Astro.props.content || Astro.props}`?

Reply:
To address the last point first: If the heading is "Frontmatter as props" then this is how the frontmatter will be used in that Astro component, both md and mdx. Yes, this Astro component *could* also be receiving regular Astro props, in which case it would need to use the trick shown in layouts. But, the heading is for passing frontmatter. I don't know if this tells us that we ALWAYS want to refer to props coming from md/mdx as `Astro.props.content` ?? 

As to whether to keep it, originally there was no code example. It just said "look up at how a layout does it." It's always an option to go back to that, and to be able to keep the nice heading that probably is what many people will be looking for.  :)

The intention here was to provide a small example within the section itself, and being smaller, could show something like a mapping of tags instead of focusing on the entire concept of a layout (which needs to be a larger code snippet if we want to keep things truly copy/pastable). Additionally, we know many readers won't read more than the section they are in, so I thought having something here might be helpful?

Reply:
Yeah, I think the issue is that there’s nothing “special” about “frontmatter as props”, which I think having a dedicated section might imply. I was reading it thinking “Ooh, a component can magically get `content`!” whereas actually the advice we’re trying to give is: “`content` is just another (not special) JavaScript object — you can pass it to a component if you like”.